bridges/TwitterBridge.php

```diff
@@ -251,94 +312,145 @@ public function collectData(){
 			}
 		}
 
+		// Filter out unwanted tweets
+		foreach ($data as $tweet) {
+			// Filter out retweets to remove possible duplicates of original tweet
+			switch($this->queriedContext) {
+			case 'By keyword or hashtag':
+				if (isset($tweet->retweeted_status) && substr($tweet->full_text, 0, 4) === 'RT @') {
+					continue 2;
+				}
+				break;
+			}
+			$tweets[] = $tweet;
+		}
+
 		$hidePictures = $this->getInput('nopic');
 
-		$promotedTweetIds = array_reduce($data->timeline->instructions[0]->addEntries->entries, function($carry, $entry) {
-			if (!isset($entry->content->item)) {
-				return $carry;
-			}
-			$tweet = $entry->content->item->content->tweet;
-			if (isset($tweet->promotedMetadata)) {
-				$carry[] = $tweet->id;
-			}
-			return $carry;
-		}, array());
+		// $promotedTweetIds = array_reduce($data->timeline->instructions[0]->addEntries->entries, function($carry, $entry) {
+		//	if (!isset($entry->content->item)) {
+		//		return $carry;
+		//	}
+		//	$tweet = $entry->content->item->content->tweet;
+		//	if (isset($tweet->promotedMetadata)) {
+		//		$carry[] = $tweet->id;
+		//	}
+		//	return $carry;
+		// }, array());
 
 		$hidePinned = $this->getInput('nopinned');
 		if ($hidePinned) {
 			$pinnedTweetId = null;
-			if (isset($data->timeline->instructions[1]) && isset($data->timeline->instructions[1]->pinEntry)) {
-				$pinnedTweetId = $data->timeline->instructions[1]->pinEntry->entry->content->item->content->tweet->id;
+			if ($user) {
+				if ($user->pinned_tweet_ids_str) {
+					$pinnedTweetId = $user->pinned_tweet_ids_str;
+				}
 			}
 		}
 
-		$tweets = array();
-
 		// Extract tweets from timeline property when in username mode
 		// This fixes number of issues:
 		// * If there's a retweet of a quote tweet, the quoted tweet will not appear in results (since it wasn't retweeted directly)
 		// * Pinned tweets do not get stuck at the bottom
-		if ($this->queriedContext === 'By username') {
-			foreach($data->timeline->instructions[0]->addEntries->entries as $tweet) {
-				if (!isset($tweet->content->item)) continue;
-				$tweetId = $tweet->content->item->content->tweet->id;
```

Comment:
Why removing this?

Reply:
I haven't found the necessary elements in tweet structure from V1.1 API so far.
This is one case, i still need to migrate or at least proof, that it really isn't necessary anymore.

Reply:
I see. I suggest to remove it including commented lines.